hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -66,21 +70,47 @@ public ClientPartitionServiceImpl(HazelcastClientInstanceImpl client) {
         clientExecutionService = (ClientExecutionServiceImpl) client.getClientExecutionService();
     }
 
-    public void start() {
-        // use internal execution service for all partition refresh process (do not use the user executor thread)
+    public void listenPartitionTable(Connection ownerConnection) {
+        //scheduling left in place to support server versions before 3.9.
         clientExecutionService.scheduleWithRepetition(new RefreshTask(), INITIAL_DELAY, PERIOD, TimeUnit.SECONDS);
+
+        //Servers after 3.9 supports listeners
+        ClientMessage clientMessage = ClientAddPartitionListenerCodec.encodeRequest();
+        ClientInvocation invocation = new ClientInvocation(client, clientMessage, ownerConnection);
+        invocation.setEventHandler(this);
+        invocation.invokeUrgent();
```

Comment:
When talking to 3.9 and later, should we fail if the invocation result is a failure? and For earlier than 3.9, we probably do not need to check the invocation result as done here.

Reply:
Yes, you are right. We need to wait for response to see if it is failed. 
But if I add I will broke the compatibility with some servers. Because we have released servers that does not return any response when client message is not supported. Related issue if you are wondering. 
https://github.com/hazelcast/hazelcast/issues/10196

Reply:
A second way can be write the code according to server version coming with response in authentication.